### PR TITLE
app-editor/{vim-core,vim,gvim}: add 9.0.0049

### PR DESCRIPTION
**Pull request by laumann**
Just some notes on differences with previous version:

 - ebuilds have all been restructured to put eapply_user first, then a slew of patching
 - all patching of source files with sed from src_configure is moved to src_prepare
 - the gentoo maintained patches now live at https://gitweb.gentoo.org/proj/vim-patches.git/?h=main

### CODE REVIEW

**Review by floppym — CHANGES_REQUESTED** [review]
> I intend to implement a vim.eclass - this package bump is just to become familiar with the ebuilds

Seems reasonable. An alternative would be to consolidate them into a single package.

> should patches be fetched via gitweb.gentoo.org? (or is there a better way to obtain a snapshot from a tag?)

Fetching from gitweb.gentoo.org is fine. The files will get mirrored anyway.

> Python 3.11 support?

Yes please.

> the blockers on older versions of vim (as I understand it: due to file collisions with files going in /usr/share), do they need to be kept/updated?

We should keep any blockers around for at least a year or two after they were first added to the ebuild. I don't think they would require any updating, unless you have shifted installed files between the 3 packages.

File: app-editors/gvim/gvim-9.0.0049.ebuild [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+	# people with broken alphabets run into trouble. bug 82186.
+	unset LANG LC_ALL
+	export LC_COLLATE="C"
+
+	use lua && lua-single_pkg_setup
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	eapply_user
```
[the remark this answers is not in the document]
Comment (laumann) [reply app-editors/gvim/gvim-9.0.0049.ebuild]: I moved eapply_user to the end of the ebuilds again after getting some input in #gentoo-dev. I considered re-ordering the sed'ing to segment into "basically a patch" and sed'ing that relies on variables and putting eapply_user inbetween, but I'd rather spend some time working on consolidating the three ebuilds into one.

Let me know if you want it eapply_user in another place.

File: app-editors/gvim/gvim-9.0.0049.ebuild [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+	# to sed this out to avoid screwiness. (1 Sep 2004 ciaranm)
+	if ! use cscope; then
+		sed -i -e \
+			'/# define FEAT_CSCOPE/d' src/feature.h || die "couldn't disable cscope"
+	fi
+
+}
+
+src_configure() {
+	local myconf=()
```
[the remark this answers is not in the document]
Comment (laumann) [reply app-editors/gvim/gvim-9.0.0049.ebuild]: done

**Comment by laumann** [comment]
> > I intend to implement a vim.eclass - this package bump is just to become familiar with the ebuilds
> 
> Seems reasonable. An alternative would be to consolidate them into a single package.

I like this idea, and will look into it tomorrow. One thing that just jumped out at me was that (g)vim has python support, but vim-core doesn't. Let's see what else I find out tomorrow.
 
> > should patches be fetched via gitweb.gentoo.org? (or is there a better way to obtain a snapshot from a tag?)
> 
> Fetching from gitweb.gentoo.org is fine. The files will get mirrored anyway.

Thanks, I'll keep it this way.
 
> > Python 3.11 support?
> 
> Yes please.

Also done.

> > the blockers on older versions of vim (as I understand it: due to file collisions with files going in /usr/share), do they need to be kept/updated?
> 
> We should keep any blockers around for at least a year or two after they were first added to the ebuild. I don't think they would require any updating, unless you have shifted installed files between the 3 packages.

I didn't change anything regarding installed files among the packages. From scanning the generated image folders after src_install, for each of the ebuilds the content lists look pretty much the same when comparing to the current stable vim8.

**Pull request closed without merging**